File: ruby/8.3.rb
#flag = 1
#i = 101
#j = 2
#while i < 200 && j <= Math.sqrt(200)
  #i += 1
  #flag = 1
    #j += 1
    #if i%j != 0
      #puts i
    #else 
      #" "
      ##break
    #end
    ##print "i", i, " "
    ##print "j", j, " "
      ##if flag == 1
        ##puts i
      #end
##end
##end
# E3.12-1.rb 求50以内的素数
$arr=[ ] #建立一个全局数组 $arr
$arr[0]=2

def add_prime(n) #定义方法 将 n以内的奇素数加入$arr
  3.step(n,2){|num|$arr <<num if is_prime?num }
end

def is_prime?(number) #定义方法 判断一个数是否是素数
  j=0 #数组下标
  while $arr[j] * $arr[j] <=number
    return false if number % $arr[j] ==0
    j +=1
  end
  return true
end

add_prime(200)
print $arr.join(", "),"\n" #转换成字符串输出

File: 8.3.c
/*
 如何求素数  1． 自然数是0，1，2……  2． 素数是2，3，5……（不包括1的只能背1和它本身整除的自然数）
*/

#include<stdio.h> 
#include <math.h> 
void main() 
{ 
  int i ,j, flag=1; 
  for(i=101; i<200; i++) 
  { 
    flag = 1;
    for(j=2; j<=sqrt(200); j++)    
      if(i%j == 0)
      { 
        flag = 0;   
        break;     
      }   
    if(flag == 1) 
      printf("i=%d是素数\n",i);
  } 
} 

/*
 bob@hp:c_and_ruby_exercises$ gcc 8.3.c -o 8.3
 bob@hp:c_and_ruby_exercises$ ./8.3
 i=101是素数
 i=103是素数
 i=107是素数
 i=109是素数
 i=113是素数
 i=127是素数
 i=131是素数
 i=137是素数
 i=139是素数
 i=149是素数
 i=151是素数
 i=157是素数
 i=163是素数
 i=167是素数
 i=173是素数
 i=179是素数
 i=181是素数
 i=191是素数
 i=193是素数
 i=197是素数
 i=199是素数
*/
